lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -71,6 +71,211 @@ describe('AwsCompileFunctions', () => {
+
+      return fixtures
+        .extend('functionLayers', {
+          layers: { testLayer: { path: 'testLayer', description: 'Different description' } },
+        })
+        .then(fixturePath => {
+          return runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          expect(firstVersionId).to.not.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create the same version ids for identical lambdas with layers with only different S3Key paths', () => {
+      // the S3Key path is timestamped and so changes on every deployment regardless of layer changes, and should
+      // therefore not be included in the version id digest
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+      const firstS3Key = firstCfTemplate.Resources.TestLayerLambdaLayer.Properties.Content.S3Key;
+      return fixtures
+        .extend('functionLayers', {})
+        .then(fixturePath => {
+          return runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          expect(firstS3Key).to.not.equal(
+            data.cfTemplate.Resources.TestLayerLambdaLayer.Properties.Content.S3Key
+          );
+          expect(firstVersionId).to.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create the same version ids for identical lambdas with differently ordered properties', () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+
+      return fixtures
+        .extend('functionLayers', {
+          functions: {
+            foo: { layers: [{ Ref: 'TestLayerLambdaLayer' }], handler: 'index.handler' },
+          },
+        })
+        .then(fixturePath => {
+          return runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+            awsRequestStubMap: {
+              CloudFormation: {
+                describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+              },
+            },
+          });
+        })
+        .then(data => {
+          expect(firstVersionId).to.equal(
+            data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+          );
+        });
+    });
+
+    it('should create different version ids if a property key changes but the value does not', () => {
```

Comment:
`should create different lambda version id for different property keys (but no different values)`